pageserver/src/tenant/timeline.rs

```diff
@@ -1994,32 +2244,26 @@ impl Timeline {
 
         // Now that we have reshuffled the data to set of new delta layers, we can
         // delete the old ones
-        let mut layer_paths_do_delete = HashSet::with_capacity(deltas_to_compact.len());
+        let mut layer_paths_to_delete = Vec::with_capacity(deltas_to_compact.len());
         drop(all_keys_iter);
         for l in deltas_to_compact {
             if let Some(path) = l.local_path() {
                 self.metrics
                     .current_physical_size_gauge
                     .sub(path.metadata()?.len());
-                layer_paths_do_delete.insert(path);
+                layer_paths_to_delete.push(path);
             }
             l.delete()?;
             layers.remove_historic(l);
         }
         drop(layers);
 
-        if self.can_upload_layers() {
-            storage_sync::schedule_layer_upload(
-                self.tenant_id,
-                self.timeline_id,
-                new_layer_paths,
-                None,
-            );
-            storage_sync::schedule_layer_delete(
-                self.tenant_id,
-                self.timeline_id,
-                layer_paths_do_delete,
-            );
+        // Also schedule the deletions in remote storage
+        if let Some(remote_client) = &self.remote_client {
+            // FIXME: This also uploads new index file. If
+            // flush_frozen_layer() is doing this at the same time, do
+            // we have a problem?
```

Comment:
I don't remember what exactly I was worried about here, but I don't see a problem now. I agree with your analysis, and I think it's harmless because the files should not intersect.